src/app/(main)/activity/[semesterId]/[activityId]/boards/[boardId]/posts/[postId]/edit/components/form/Form.tsx

```diff
@@ -0,0 +1,128 @@
+'use client'
+
+import { useEffect } from 'react'
+import { useForm } from 'react-hook-form'
+
+import { zodResolver } from '@hookform/resolvers/zod'
+import { useMutation, useQuery } from '@tanstack/react-query'
+import { useParams, usePathname, useRouter } from 'next/navigation'
+
+import { PostContentFieldEditor } from '@/components/feature/post/create-post-form/editor'
+import {
+  Button,
+  Form,
+  FormControl,
+  FormField,
+  FormItem,
+  FormMessage,
+  Input,
+  Label,
+  Separator,
+  useToast,
+} from '@/components/ui'
+import { queryClient } from '@/lib/query-client'
+import { CreateActivityPost, CreateActivityPostSchema } from '@/schema/post'
+import { activityPostQuries, updateActivityPostApi } from '@/service/api'
+
+import { ActivityDateFieldDialog } from './date-field-dialog'
+
+interface EditActivityPostFormProps {
+  boardId: number
+}
+
+export const EditActivityPostForm = ({
+  boardId,
+}: EditActivityPostFormProps) => {
+  const { toast } = useToast()
+  const router = useRouter()
+  const pathName = usePathname()
+  const params = useParams()
+
+  const postId = Number(params.postId)
+
+  const { data: editPostData } = useQuery(
+    activityPostQuries.detail({ boardId, postId }),
+  )
+
+  const { mutate: updateActivityPost, isPending } = useMutation({
+    mutationFn: updateActivityPostApi,
+    onSuccess: (data) => onSuccess(data.message),
+  })
+
+  const form = useForm<CreateActivityPost>({
+    resolver: zodResolver(CreateActivityPostSchema),
+    defaultValues: {
+      postTitle: '',
+      postContent: '',
+      postImageIds: [],
+      postActivityStartDate: '',
+      postActivityEndDate: '',
+    },
+  })
+
+  useEffect(() => {
+    if (editPostData) {
+      form.reset({
+        postTitle: editPostData.postTitle,
+        postContent: editPostData.postContent,
+        postActivityStartDate: editPostData.postActivityStartDate,
+        postActivityEndDate: editPostData.postActivityEndDate,
+      })
+    }
+  }, [editPostData, form])
+
+  const onSuccess = (message?: string) => {
+    toast({
+      title: message,
+      duration: 2000,
+    })
+
+    queryClient.refetchQueries({
+      queryKey: activityPostQuries.lists(boardId),
+    })
+
+    const basePath = pathName.split('/').slice(0, -1).join('/')
+    router.push(basePath)
+  }
+
+  return (
+    <Form {...form}>
+      <form
+        onSubmit={form.handleSubmit((values) =>
+          updateActivityPost({ boardId, postId, data: values }),
+        )}
```

Comment:
다음엔 아래 코드와 같이 작성해서 디버깅해보세요~ 디버깅 한 후에는 error 부분 삭제하고 커밋 날려야합니다

```suggestion
onSubmit={form.handleSubmit(
          (values) => updateActivityPost({ boardId, postId, data: values }),
          (error) => console.log(error),
        )}
```
<img width="376" alt="스크린샷 2025-01-13 오후 1 54 40" src="https://github.com/user-attachments/assets/f038cf45-80a7-4361-9a80-96f6da6ac804" />

- postActivityEndDate, postImageIds에서 zod 유효성 검사 에러가 발생하네요